docs/source/models.rst

```diff
@@ -81,6 +81,141 @@ time_created
+Model signatures are recognized and enforced by standard :ref:`MLflow model deployment tools
+<built-in-deployment>`. For example, the :ref:`mlflow models serve <local_model_deployment>` tool,
+which deploys a model as a REST API, validates inputs based on the model's signature.
+
+The following example displays an excerpt from MLmodel file with the model signature for a
+classification model trained on the `Iris dataset <https://archive.ics.uci.edu/ml/datasets/iris>`_.
+The input has 4 named, numeric columns. The output is an unnamed integer specifying the predicted
+class:
+
+.. code-block:: yaml
+
+  signature:
+      inputs: '[{"name": "sepal length (cm)", "type": "double"}, {"name": "sepal width
+        (cm)", "type": "double"}, {"name": "petal length (cm)", "type": "double"}, {"name":
+        "petal width (cm)", "type": "double"}]'
+      outputs: '[{"type": "integer"}]'
+
+Signature Enforcement
+~~~~~~~~~~~~~~~~~~~~~
+When scoring a model that includes a signature, inputs are validated based on the signature's input
+schema. This input schema enforcement checks input column ordering and column types, raising an
+exception if the input is not compatible. This enforcement is applied in MLflow before calling the
+underlying model implementation. Note that this enforcement only applies when using :ref:`MLflow
+model deployment tools <built-in-deployment>` or when loading models as ``python_function``. In
+particular, it is not applied to models that are loaded in their native format (e.g. by calling
+:py:func:`mlflow.sklearn.load_model() <mlflow.sklearn.load_model>`).
+
+Column Ordering Enforcement
+"""""""""""""""""""""""""""
+The input columns are checked against the model signature. If there are any missing columns,
+MLflow will raise an exception. Extra columns that were not declared in the signature will be
+ignored. If the input schema in the signature defines column names, column matching is done by name
+and the columns are reordered to match the signature. If the input schema does not have column
+names, matching is done by position (i.e. MLflow will only check the number of columns).
+
+Column Type Enforcement
+"""""""""""""""""""""""
+The input column types are checked against the signature. MLflow will perform safe type conversions
+if necessary. Generally, only upcasts (e.g. integer -> long or float -> double) are considered to be
+safe. If the types cannot be made compatible, MLflow will raise an error.
+
+How To Log Models With Signatures
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To include signature with your model, pass :py:class:`signature object
+<mlflow.models.ModelSignature>` as an argument to the appropriate log_model call, e.g.
+:py:func:`sklearn.log_model() <mlflow.sklearn.log_model>`. The model signature object can be created
+by hand or :py:func:`inferred <mlflow.models.infer_signature>` from datasets with valid model input
```

Comment:
I still think `input` should be made plural here. The reason is that training datasets usually contain many model inputs:

```suggestion
by hand or :py:func:`inferred <mlflow.models.infer_signature>` from datasets with valid model inputs
```